Touch up image-trust man

TomSweeneyRedHat opened the pull request:
Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

In #1899, @rhatdan asked me to try and tweak the verbiage
a bit in the podman-images-trust man page.  I've done so,
or attempted to, and have also added a policy-json(5) link.

rhatdan commented on docs/podman-image-trust.1.md:
```diff
@@ -1,7 +1,7 @@
 
 # NAME
-podman\-trust - Manage container registry image trust policy
+podman\-trust - Manage container image trust policy
```
Why are you removing registry.  podman trust is identifying whether you trust registry images, ones on your current system are not checked?

TomSweeneyRedHat replied:
I'll add it back in.  I'd thought it was redundant, but had thought local vs registry images.

rhatdan commented on docs/podman-image-trust.1.md:
```diff
@@ -14,12 +14,12 @@ podman\-trust - Manage container registry image trust policy
-Manages the trust policy of the host system. Trust policy describes
-a registry scope (registry and/or repository) that must be signed by public keys. Trust
-is defined in **/etc/containers/policy.json**. Trust is enforced when a user attempts to pull
+Manages the trusted keys for a container image usually dependent on its location.
```
We manage other things the keys.  We setup which registiry and which repository are trusted.  Talking about secrets keys is really selling this short.

TomSweeneyRedHat replied:
OK, I've tried to rework it, if my latest isn't on point, please suggest further verbiage.

rhatdan commented:
Lets through it into a google doc so we can work on it together.

TomSweeneyRedHat commented:
@rhatdan @vrothberg I've updated the PR, I made took all of the open comments and made a few minor tweaks (extra spaces, missing dash, removed a duplicate word).  I think this is gtg.

rhatdan commented:
/lgtm
